feature_extractor.py

```diff
@@ -132,12 +132,13 @@ def read_features(file_path, cache=None):
     features = None
     with open(file_path, 'r') as fp:
         data = fp.read().splitlines(keepends=False)
-        features = np.zeros((len(data), 4096))
+        features = np.zeros((len(data), feature_dim))
         for i, line in enumerate(data):
             features[i, :] = [float(x) for x in line.split(' ')]
 
     features = torch.from_numpy(features).float()
-    cache[file_path] = features
+    if cache is not None:
+        cache[file_path] = features
```

Comment:
This because I commited as "change" and reset it.
I didn't know that you committed the name "change" before.